pkg/await/apps_deployment.go

```diff
@@ -394,6 +395,23 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 		// No current generation, Deployment controller has not yet created a ReplicaSet. Do
 		// nothing.
 		return
+	} else {
+		if currentGenerationInt, err := strconv.Atoi(dia.currentGeneration); err == nil {
+			if int64(currentGenerationInt) != dia.deployment.GetGeneration() {
+				// If the generation is set, make sure it matches the revision annotation, otherwise, ignore this
+				// event because the status we care about may not be set yet.
+				return
+			}
+			if rawObservedGeneration, ok := openapi.Pluck(
```

Comment:
Is there a particular reason why we're starting with the metadata and falling back to the status, rather than the reverse?

Reply:
No; I just wanted to check both conditions, but the order wasn't material.